### PR TITLE
DCP 549: Overwrite project on spreadsheet upload

### PR DESCRIPTION
**Pull request by aaclan-ebi**
ebi-ait/dcp-ingest-central#549

Pass the updateProject user preference to the request

Note: need to adjust the integration tests

### CODE REVIEW

File: broker/service/spreadsheet_upload_service.py [hunk 1]
```diff
@@ -18,7 +18,12 @@ def __init__(self, ingest_api: IngestApi, storage_service: SpreadsheetStorageSer
         self.storage_service = storage_service
         self.importer = importer
 
-    def async_upload(self, token, request_file, is_update, project_uuid=None, submission_uuid=None):
+    def async_upload(self, token, request_file, params):
+        project_uuid = params.get('projectUuid')
+        submission_uuid = params.get('submissionUuid')
+        is_update = params.get('isUpdate')
```
Comment (ke4) [inline broker/service/spreadsheet_upload_service.py:24]: What happens if these parameters are not provided in the `params` dictionary? Previously `is_update` did not have a default value.

**Review by ke4 — COMMENTED** [review]
I added a comment, could you review it, please?

File: broker/service/spreadsheet_upload_service.py (hunk 1 above)
Comment (aaclan-ebi) [reply broker/service/spreadsheet_upload_service.py:24]: It's okay for all these params to be None. They are all optional. In line 34 below, the `if is_update` condition will just evaluate to false.

Comment (aaclan-ebi) [reply broker/service/spreadsheet_upload_service.py:24]: The flag `is_update` is for importing a spreadsheet with UUIDs and updates. If it's falsy like None, it means it's a new submission and `project_uuid` and `update_project` will indicate that there is an existing project and the user preference to update it or not.

**Review by MightyAx — APPROVED** [review]

**Review by ke4 — APPROVED** [review]

**Pull request merged**